scripts/getresdata.py

```diff
@@ -0,0 +1,179 @@
+
+public:
+  ResidueData() {}
+  ResidueData(std::string name, std::vector<std::string> atomNames,
+              std::vector<std::pair<std::string, std::string>> singleBonds,
+              std::vector<std::pair<std::string, std::string>> doubleBonds)
+  {
+    m_residueName = name;
+    m_residueAtomNames = atomNames;
+    m_residueSingleBonds = singleBonds;
+    m_residueDoubleBonds = doubleBonds;
+  }
+
+  ResidueData(const ResidueData& other)
+  {
+    m_residueName = other.m_residueName;
+    m_residueAtomNames = other.m_residueAtomNames;
+    m_residueSingleBonds = other.m_residueSingleBonds;
+    m_residueDoubleBonds = other.m_residueDoubleBonds;
+  }
+
+  ResidueData& operator=(ResidueData other)
+  {
+    using std::swap;
+    swap(*this, other);
+    return *this;
+  }
+
+  std::vector<std::pair<std::string, std::string>> residueSingleBonds()
+  {
+    return m_residueSingleBonds;
+  }
+
+  std::vector<std::pair<std::string, std::string>> residueDoubleBonds()
+  {
+    return m_residueDoubleBonds;
+  }
+};
+'''
+)
+
+final_ligands = []
+for ligand in ligands:
+    sdf = requests.get(sdfTemplate.format(ligand[0], ligand, ligand))
+    # there *must* be a way to do this from a requests buffer, but this works
+    with open('temp.sdf', 'wb') as handle:
+        for block in sdf.iter_content(1024):
+            handle.write(block)
+
+    mol_sdf = next(pybel.readfile("sdf", 'temp.sdf'))
+    if len(mol_sdf.atoms) < 2:
+        continue
+    final_ligands.append(ligand)
+
+    pdb = requests.get(pdbTemplate.format(ligand[0], ligand, ligand))
+    with open('temp.pdb', 'wb') as handle:
+        for block in pdb.iter_content(1024):
+            handle.write(block)
+
+    mol_pdb = next(pybel.readfile("pdb", 'temp.pdb'))
+    atom_map = {}
+    for i in range(len(mol_sdf.atoms)):
+        idx = mol_sdf.atoms[i].idx
+        atom = mol_pdb.atoms[i].OBAtom
+        res = atom.GetResidue()
+        # build up a map between atom index and atom ID
+        atom_map[idx] = res.GetAtomID(atom).strip().rstrip()
+
+    # go through bonds
+    single_bonds = []
+    double_bonds = []
+    for bond in ob.OBMolBondIter(mol_sdf.OBMol):
+        begin = bond.GetBeginAtomIdx()
+        end = bond.GetEndAtomIdx()
+        if bond.GetBO() == 2:
+            double_bonds.append((atom_map[begin], atom_map[end]))
+        elif bond.GetBO() == 1:
+            single_bonds.append((atom_map[begin], atom_map[end]))
+
+    # print out the residue data
+    print('ResidueData %sData("%s",' % (ligand, ligand))
+    print('// Atoms')
+    print('{')
+    for atom in atom_map.values():
+        print('"%s", ' % atom)
+    print('},')
+
+    print('// Single Bonds')
+    print('{')
+    for bond in single_bonds:
+        print('{ "%s", "%s" },' % bond)
+    print('},')
+
+    print('// Double Bonds')
+    print('{')
+    if len(double_bonds):
+        for bond in double_bonds:
+            print('{ "%s", "%s" },' % bond)
+    print('},')
```

Comment:
Comma is unnecessary, though semantically there's nothing wrong with it.

Reply:
I think I have a slightly different version on my office computer. Thanks for the heads-up.